share/smack/lib/smack.c

```diff
@@ -2053,18 +2053,17 @@ void __SMACK_decls() {
   D("procedure {:inline 1} $$alloc(n: ref) returns (p: ref)\n"
     "modifies $Alloc, $CurrAddr;\n"
     "{\n"
-    "  p := $CurrAddr;\n"
-    "  havoc $CurrAddr;\n"
+    "  assume $sge.ref.bool(n, $0.ref);\n"
```

Comment:
In some situations when unbounded integers are used, yes, I believe you are right. But that is a more general issue related to our inability to correctly model unsigned integers (and casts between unsigned and signed) when unbounded integers are used. So I can create an example even without malloc where a bug would be missed due to this deficiency - if I remember a common patters we do not model correctly is casting a negative integer into an unsigned integer.